fe/fe-core/src/main/java/com/starrocks/scheduler/PartitionBasedMaterializedViewRefreshProcessor.java

```diff
@@ -356,6 +360,18 @@ private void updateMetaForOlapTable(MaterializedView.AsyncRefreshContext refresh
                     currentVersionMap.get(tableId);
             Map<String, MaterializedView.BasePartitionInfo> partitionInfoMap = tableEntry.getValue();
             currentTablePartitionInfo.putAll(partitionInfoMap);
+            Table table = tableEntry.getKey().getTable();
+            Preconditions.checkState(table.isOlapTable());
+            OlapTable olapTable = (OlapTable) table;
+            // set partitions that exist in base but not read for mv, which may be pruned by predicates.
+            // used to judge whether these partition are used by mv.
+            // if the partition version is -1, it means that the partition is not used by mv.
+            // when decide whether mv should be refresh, these partition are excluded to consider
+            for (Partition partition : olapTable.getPartitions()) {
+                MaterializedView.BasePartitionInfo invalidPartitionInfo =
+                        new MaterializedView.BasePartitionInfo(partition.getId(), -1);
+                currentTablePartitionInfo.putIfAbsent(partition.getName(), invalidPartitionInfo);
+            }
```

Comment:
It's weird to  add an invalid  here. Why not mock this in `getUpdatedPartitionNamesOfOlapTable` instead of `updateMetaForOlapTable `